nvflare/app_opt/flower/applet.py

```diff
@@ -26,6 +26,25 @@
 from nvflare.security.logging import secure_format_exception
 
 
+def get_partition_id(fl_ctx: FLContext):
+    """Get the partition id for the current client based on the sorted list of all client names."""
+    engine = fl_ctx.get_engine()
+
+    all_client_names = sorted([client.name for client in engine.all_clients])
+
+    for id, client_name in enumerate(all_client_names):
+        if client_name == fl_ctx.get_identity_name():
+            return id
+
+    return -1
+
+
+def get_num_partitions(fl_ctx: FLContext):
+    """Get the number of partitions based on the number of clients."""
+    engine = fl_ctx.get_engine()
+    return len(engine.all_clients)
```

Comment:
instead of accessing instance variable "all_clients", let's use the interface -> `engine.get_clients()`

Reply:
Good point. I changed to get_clients().